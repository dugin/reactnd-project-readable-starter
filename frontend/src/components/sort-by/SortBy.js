import React, {Component} from 'react';
import {ButtonDropdown, DropdownToggle, DropdownMenu, DropdownItem} from 'reactstrap';
import './SortBy.css';
import propTypes from 'prop-types';

class SortBy extends Component {

    constructor(props) {
        super(props);

        this.state = {
            dropdownOpen: false,
            selected: props.default
        };
    }

    componentWillReceiveProps(nextProps){
        this.setState({selected: this.state.selected  ||  nextProps.default  || 'category'});
    }

    setSelected = (selected) => {
        this.setState({selected}, () => this.props.selected(selected));
    };

    toggle = () => {
        this.setState({
            dropdownOpen: !this.state.dropdownOpen
        });
    };

    render() {
        return (
            <ButtonDropdown className="SortBy" isOpen={this.state.dropdownOpen} toggle={this.toggle}>
                <DropdownToggle caret>
                    {this.state.selected}
                </DropdownToggle>
                <DropdownMenu>
                    {this.props.options.map(s =>
                        (<DropdownItem onClick={() => this.setSelected(s)} key={s}> {s}</DropdownItem>)
                    )}

                </DropdownMenu>
            </ButtonDropdown>
        )
    }
}

SortBy.propTypes = {
    selected: propTypes.func.isRequired,
    options: propTypes.array.isRequired,
    default: propTypes.string
};


export default SortBy;