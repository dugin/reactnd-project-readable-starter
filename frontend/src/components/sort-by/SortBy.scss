@import "../../variables";

.SortBy{
  width: 100%;
  display: flex;
  flex-direction: row-reverse;
  border-bottom: 1px solid $line-color;

  .dropdown-toggle{
    background-color: transparent;
    border: none;
    color: $gray-700;

    &:focus, &:hover{
      background-color: transparent;
      border: none;
      box-shadow: none;
    }
  }
}