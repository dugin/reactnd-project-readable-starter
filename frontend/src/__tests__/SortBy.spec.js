import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import SortBy from "../components/sort-by/SortBy";

describe('SortBy Component', () => {

    const options = ['react', 'redux', 'angular'];

    const sortBy = shallow(<SortBy selected={jest.fn()} options={options}/>);

    it('renders properly', () => {

        expect(toJson(sortBy)).toMatchSnapshot();

    });
});