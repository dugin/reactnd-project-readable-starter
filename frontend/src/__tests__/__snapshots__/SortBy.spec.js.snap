// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBy Component renders properly 1`] = `
<ButtonDropdown
  className="SortBy"
  isOpen={false}
  toggle={[Function]}
>
  <DropdownToggle
    aria-haspopup={true}
    caret={true}
    color="secondary"
  >
    react
  </DropdownToggle>
  <DropdownMenu
    flip={true}
    tag="div"
  >
    <DropdownItem
      key="react"
      onClick={[Function]}
      tag="button"
      toggle={true}
    >
       
      react
    </DropdownItem>
    <DropdownItem
      key="redux"
      onClick={[Function]}
      tag="button"
      toggle={true}
    >
       
      redux
    </DropdownItem>
    <DropdownItem
      key="angular"
      onClick={[Function]}
      tag="button"
      toggle={true}
    >
       
      angular
    </DropdownItem>
  </DropdownMenu>
</ButtonDropdown>
`;
